src/BlazorWasmSdk/Targets/Microsoft.NET.Sdk.BlazorWebAssembly.6_0.targets

```diff
@@ -115,12 +124,75 @@ Copyright (c) .NET Foundation. All rights reserved.
+        <CopyLocal>true</CopyLocal>
+      </_DotNetJsItem>
+    </ItemGroup>
+
     <ItemGroup>
+      <_DotnetJsStaticWebAssetCandidate Remove="@(_DotnetJsStaticWebAssetCandidate)" />
+      <_DotnetJsCopyCandidates Remove="@(_DotnetJsCopyCandidates)" />
+    </ItemGroup>
+
+    <DefineStaticWebAssets Condition="'@(_DotNetJsItem->Count())' != '0'"
+      CandidateAssets="@(_DotNetJsItem)"
+      SourceId="$(PackageId)"
+      SourceType="Computed"
+      AssetKind="Build"
+      AssetRole="Primary"
+      AssetTraitName="BlazorWebAssemblyResource"
+      AssetTraitValue="native"
+      CopyToOutputDirectory="PreserveNewest"
+      CopyToPublishDirectory="Never"
+      ContentRoot="$(IntermediateOutputPath)\blazor"
+      BasePath="$(StaticWebAssetBasePath)"
+    >
+      <Output TaskParameter="Assets" ItemName="_DotnetJsStaticWebAssetCandidate" />
+      <Output TaskParameter="CopyCandidates" ItemName="_DotnetJsCopyCandidates" />
+    </DefineStaticWebAssets>
+
+    <Copy
+      SourceFiles="@(_DotnetJsCopyCandidates)"
+      DestinationFiles="@(_DotnetJsCopyCandidates->'%(TargetPath)')"
+      SkipUnchangedFiles="true"
+      OverwriteReadOnlyFiles="$(OverwriteReadOnlyFiles)" />
+
+    <ItemGroup Condition="'@(_DotNetJsItem->Count())' != '0'">
+      <ReferenceCopyLocalPaths Remove="@(_DotNetJsItem)" />
+      <_DotnetJsStaticWebAsset Include="@(_DotnetJsStaticWebAssetCandidate->'%(ContentRoot)_framework\dotnet.js')" />
+      <_BlazorStaticWebAsset Include="@(_DotnetJsStaticWebAsset)" />
+    </ItemGroup>
+  </Target>
+
+  <Target Name="_ScrambleDotnetJsFileName" AfterTargets="$(_ScrambleDotnetJsFileNameAfterThisTarget)">
+    <!--
+      We want the dotnet.js file output to have a version to better work with caching. We'll append the runtime version to the file name as soon as file has been discovered.
+    -->
+    <PropertyGroup>
+      <_DotNetJsVersion>$(BundledNETCoreAppPackageVersion)</_DotNetJsVersion>
+      <_DotNetJsVersion Condition="'$(RuntimeFrameworkVersion)' != ''">$(RuntimeFrameworkVersion)</_DotNetJsVersion>
+      <_BlazorDotnetJsFileName>dotnet.$(_DotNetJsVersion).js</_BlazorDotnetJsFileName>
+    </PropertyGroup>
+
+    <ItemGroup Condition="@(WasmNativeAsset->Count()) == 0">
       <_DotNetJsItem Include="@(ReferenceCopyLocalPaths)" Condition="'%(ReferenceCopyLocalPaths.DestinationSubPath)' == 'dotnet.js' AND '%(ReferenceCopyLocalPaths.AssetType)' == 'native'">
         <RelativePath>_framework/$(_BlazorDotnetJsFileName)</RelativePath>
       </_DotNetJsItem>
     </ItemGroup>
 
+    <ItemGroup Condition="@(WasmNativeAsset->Count()) != 0">
+      <_DotNetJsItem Remove="@(_DotNetJsItem)" />
+      <_DotNetJsItem Include="@(WasmNativeAsset)" Condition="'%(WasmNativeAsset.FileName)%(WasmNativeAsset.Extension)' == 'dotnet.js'">
+        <RelativePath>_framework/$(_BlazorDotnetJsFileName)</RelativePath>
+        <AssetType>native</AssetType>
+        <CopyLocal>true</CopyLocal>
+      </_DotNetJsItem>
+    </ItemGroup>
+
```

Comment:
Based on offline discussion, we don't need the scrambling here since @TanayParikh took care of it on his PR

Reply:
Does that mean that the two targets `_Scramble*` can be dropped completely?

Reply:
Ah, the target is no longer there! 👍

Reply:
Yep. We do this all the time now on build and publish if necessary, as it improves the cache busting